be/dtm0_log.h

```diff
@@ -0,0 +1,235 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+#pragma once
+#ifndef __MOTR_BE_DTM0_LOG_H__
+#define __MOTR_BE_DTM0_LOG_H__
+
+#include "be/list.h"		/* m0_be_list */
+#include "dtm0/tx_desc.h"	/* m0_dtm0_tx_desc */
+#include "fid/fid.h"		/* m0_fid */
+#include "lib/buf.h"		/* m0_buf */
+#include "dtm0/dtx.h"           /* struct m0_dtm0_dtx */
+
+struct m0_be_tx;
+struct m0_be_tx_credit;
+struct m0_dtm0_clk_src;
+
+/**
+ *  @page dtm0 log implementation
+ *
+ *  @section Overview
+ *  DTM0 log module will be working on incoming message request, the goal of
+ *  this module is to track journaling of incoming message either on persistent
+ *  or volatile memory based on whether logging is happening on participant or
+ *  on originator, so that in the phase failure consistency of failed
+ *  participant can be restored by iterating over logged journal(DTM0 log
+ *  record) to decide which of the logged record needs to be sent as a redo
+ *  request.
+ *
+ *  A distributed transaction(dtx) has a group of states of each individual
+ *  participant of the distributed transaction(dtx) (note: originator is not a
+ *  part of this group). Each entry of such a group may have many different
+ *  states, but this set of states must include the following states:
+ *
+ *	1) InProgress - the node (owner of the log) has a request.
+ *	2) Executed - one or more operations have been executed in volatile
+ *	              memory of the participant, the result of such execution
+ *	              is known and returned to the dtx user.
+ *	3) Persistent - locally persisted in transactional context.
+ *	4) Stable - sufficient number of sent operations have been “persisted”
+ *	            on the remote end which guarantees survivial of persistent
+ *	            failures.
+ *	5) DONE -  all sent operations have been “persisted” on sufficient
+ *	           number of non-failed participants.
+ *
+ *  Every participant maintains the journal record(DTM0 Log record) that
+ *  corresponds to each distributed transaction(dtx) and it can be described by
+ *  "struct m0_dtm0_log_rec" which will be stored in volatile/persistent
+ *  storage. Basically this log record will maintain txr id, group of state of
+ *  each individual participant and payload.
+ *
+ *  When a distributed transaction(dtx) is executed by participant, participant
+ *  modify its state in group of state as M0_DTPS_EXECUTED and will add/modify
+ *  DTM0 log record, for rest of the participant it will keep the state as it is
+ *  in distributed transaction(dtx).
+ *
+ *  When a distributed transaction(dtx) become persistent on participant,
+ *  participant will modify its state in group of state as M0_DTPS_PERSISTENT
+ *  and will modify DTM0 log record, for rest of the participant it will keep
```

Comment:
s/modify DTM0 log record, for/modify DTM0 log record. For

These are two complete sentences.

Reply:
Done.